misk/src/main/kotlin/misk/environment/Deployment.kt

```diff
@@ -0,0 +1,39 @@
+package misk.environment
+
+/** Deployment describes the context in which the application is running */
+data class Deployment(
```

Comment:
Also thinking about

```
// Raw environment name
data class Env(val name: String)
data class Deployment(
  val name: String,

  val environment: Env, // Include raw env in Deployment

  // ...
}
```

Reply:
What would it be used for? And can we create something more formal for that?

Reply:
Uses are atypical. Some concrete examples in our internal code:
- map of env to top-level domains. The is where the low-level env like "platform-staging" vs "staging" is relevant.
- in platform-staging environment, fake out some dependencies that aren't available as they are in staging or prod.

WDYT about keeing the `Env` data class above but not including it in the `Deployment`. It can be passed around separately or bound using Guice as needed.

Reply:
This feels very skimmy to me

Reply:
Or perhaps, things that should be in config files.

Reply:
yea I think everything `Env` specific should be in a config. To me the only drawback to a config is yaml kind of sucks, especially for something that is truly global and static (e.g top-level domains). It would be nice if components could register static configs in code per `Env` and they would get magically merged with the yaml configs per `Env`. App code would just see a single unified `Config`.

Until we build something like that (or never cause it's too magical), we should put all `Env` differences in a yaml.

Reply:
thinking about it more, we already kind of have yaml + code defaults being merged. we should definitely consider expanding that to allow `Env` defaults in code.